google/cloud/ndb/query.py

```diff
@@ -704,17 +708,17 @@ def _to_filter(self, post=False):
                 representation of the filter.
 
         Raises:
-            NotImplementedError: If the ``opsymbol`` is ``!=`` or ``in``, since
-                they should correspond to a composite filter. This should
+            NotImplementedError: If the ``opsymbol`` is ``!=``, since
+                it should correspond to a composite filter. This should
                 never occur since the constructor will create ``OR`` nodes for
-                ``!=`` and ``in``
+                ``!=``.
```

Comment:
It seems like we will still create OR nodes if the list of IN values is >= 31? Or am I misunderstanding?

Reply:
Please take a look at https://github.com/googleapis/python-ndb/pull/954 instead.  I think this is a better short term improvement.